### ISSUE TITLE
Lab 3 generate waiting Event

### ISSUE THREAD

**Issue by allnotavailable** [nus-cs2030/2223-s2#299]
Hi does anyone have any tips or hints on how to emulate the waiting event queue? I have been stuck in a while.  Thank you.

**Comment by tristantanjh** [comment]
you can try using an imlist as an attribute for your servers class and add objects to it as a que :)

**Comment by ianwonghanli** [comment]
You the queue as an ImList<Customer> variable in the server class. Then, using a FIFO stack model (First in First out model), add each wait event's customer to the end of the ImList. 

And then when you want to iterate through a server's queue, check if the index Wait Event's customer is in the first index of 
the ImList<Customer> variable in the server. If the first index, remove the customer from the Queue and update the Wait Event to a Serve Event.

**Comment by notsushiii** [comment]
Personally, I didn't use any ImList or PQ. I had an integer attribute in the Server class to keep track of the number of customers in the queue and also methods to add/remove from the queue (+1/-1 to the number).

**Comment by aowz** [comment]
I used an integer to track the queue, with multiple server functions to increase or decrease the queue count.

**Comment by wmingchuan** [comment]
I used a PQ, ImList will work also. Java's implementation of Queue and LinkedList are not allowed.

**Comment by youthfrost** [comment]
yup I just used an integer to update the queue number and have a server and customer allocated to the waiting event to keep track of which customer is waiting at which server.

**Comment by Aron113** [comment]
For me, I created a Checker event that will trigger Serve event only when the customer is next in queue

**Comment by NeoHW** [comment]
You could try to implement an ImList of customers in each server and work from there

**Comment by kmkmkmkmkmkmmm** [comment]
I think giving each server an ImList<Customer> will make it easy to access the customers' information. Some of my friends used a counter for each server to check the queue size, while the customer is attached to a serverNo. that either served or is going to serve them.

**Comment by Yew-Wei** [comment]
You could use an integer to update the customer queue in the waiting event, and update the queue in the respective events itself.

**Comment by EvanCh08** [comment]
You can use your wait class as a class that contains all the waiting customers, then you can call the matching server id when Done event is executed from your data structure.

**Comment by subhadhra-m** [comment]
you can create a new event class called checkevent that will check whether the the next customer in the queue of the waitevent is the same as the current customer. If it is, then you can proceed to the serve event. If it is not, then return another check event, until the customer is the same. You can create an ImList called Queue in your server class and then add customers to that queue.

**Comment by littlemidget3** [comment]
What you could is to create an ImList of Customers in your server class to keep track of the queue. With this attribute, you could simplify your work with methods that checks the size of your server's queue, remove the first customer from the queue, or even add customers to the back of the queue ! apologies for the late comment
